addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4269,10 +4269,15 @@ export class OdooEditor extends EventTarget {
             if (fragment.hasChildNodes()) {
                 this._applyCommand('insert', fragment);
             }
-        } else if (files.length && targetSupportsHtmlContent) {
-            this.addImagesFiles(files).then(html => this._applyCommand('insert', this._prepareClipboardData(html)));
-        } else if (clipboardHtml && targetSupportsHtmlContent) {
-            this._applyCommand('insert', this._prepareClipboardData(clipboardHtml));
+        } else if ((files.length || clipboardHtml) && targetSupportsHtmlContent) {
+            // Differentiate or choose between images and html
+            const clipboardElem = document.createElement('template');
+            clipboardElem.innerHTML = this._prepareClipboardData(clipboardHtml);
```

Comment:
This produces a different behavior than in v15
![Screenshot from 2023-04-12 21-50-33](https://user-images.githubusercontent.com/92587669/231569606-20a63d75-bd9f-4cec-a4b3-e475c24bc7c3.png)

Reply:
So it's completely broken right now in this PR, right ?

Note that there is a very good reason why the new return value from `_prepareClipboardData` ends up like this when given to `innerHTML`.

Hint: Did you compare the return value from `_prepareClipboardData` in 15.0 vs master ? How is it different than the return value from 15.0 ? When you have found that, think about `innerHTML`. What is it supposed to be ?

Reply:
The return value of `_prepareClipboardData` in `master` cannot be used directly with `innerHTML` anymore because it returns actual nodes rather than a string. Sure, you can convert the nodes into string but that's not really interesting to do so because `innerHTML` will just convert the string back to nodes, so why double convert while you already have nodes ? Just to use `innerHTML` ? Then maybe it's using `innerHTML` that is a bad idea, don't you think ?

We have some nodes and we want the content of another node to equal the nodes that we have. What function could we use to do so ? Hint: we use this function at multiple places in the `insert` command.

Reply:
I meant `replaceChildren` indeed but you're right in fact: we can just call `insert` at this point. 👍